File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOMusicOperateModel.kt
//
//  IDOMusicOperateModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Operation for songs or folders event

open class IDOMusicOperateModel(operateType: Int, version: Int, errCode: Int, musicId: Int?) :
    IDOBaseModel {

    /// Operation type:
    /// 0: Invalid operation
    /// 1: Delete music
    /// 2: Add music
    /// 3: Delete folder
    /// 4: Add folder
    /// 5: Modify playlist
    /// 6: Import playlist
    /// 7: Delete music in playlist
    @SerializedName("operate_type")
    var operateType: Int = operateType

    /// Firmware SDK card information
    /// Total space
    @SerializedName("version")
    var version: Int = version

    /// 0: Successful; non-zero: Failed
    @SerializedName("err_code")
    var errCode: Int = errCode

    /// Music id returned when adding music successfully
    @SerializedName("music_id")
    var musicId: Int? = musicId

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOMusicOperateModel.swift
//
//  IDOMusicOperateModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Operation for songs or folders event
@objcMembers
public class IDOMusicOperateModel: NSObject, IDOBaseModel {
    /// Operation type:
    /// 0: Invalid operation
    /// 1: Delete music
    /// 2: Add music
    /// 3: Delete folder
    /// 4: Add folder
    /// 5: Modify playlist
    /// 6: Import playlist
    /// 7: Delete music in playlist
    public var operateType: Int
    /// Firmware SDK card information
    /// Total space
    public var version: Int
    /// 0: Successful; non-zero: Failed
    public var errCode: Int
    /// Music id returned when adding music successfully
    public var musicId: Int?
    
    enum CodingKeys: String, CodingKey {
        case operateType = "operate_type"
        case version = "version"
        case errCode = "err_code"
        case musicId = "music_id"
    }
    
    public init(operateType: Int,version: Int,errCode: Int,musicId: Int?) {
        self.operateType = operateType
        self.version = version
        self.errCode = errCode
        self.musicId = musicId
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

